### PR TITLE
feat(registry): add erg

### PR DESCRIPTION
**Pull request by pysan3**
Hi, thanks as always of your amazing work!

I am the author of #503

As noted in [ELS README](https://github.com/erg-lang/erg-language-server#els-erg-language-server), the language server has been integrated into the main compiler.

Therefore, I have changed the installation method of els to get the executable from [erg compiler](https://github.com/erg-lang/erg#flags) to be used as `erg --language-server` instead of the old method `els`.

I have already created a PR to nvim-lspconfig (https://github.com/neovim/nvim-lspconfig/pull/2404)

Thanks in advance,
@pysan3

### CODE REVIEW

**Review by williamboman — COMMENTED** [review]
Cool! However this effectively is an entirely new package, so retrofitting an old one is not the best way to go. Generally, Mason doesn't yet support managing runtimes/compilers such as `erg`, even if they bundle a language server. The reasoning is that you most likely already manage it elsewhere, so providing installation via Mason is of questionable value and might even be detrimental.

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/lua/mason-registry/erg-language-server/init.lua b/lua/mason-registry/erg-language-server/init.lua
index 92ed0bc..4bbae8a 100644
--- a/lua/mason-registry/erg-language-server/init.lua
+++ b/lua/mason-registry/erg-language-server/init.lua
@@ -6,18 +6,20 @@ local _ = require "mason-core.functional"
 return Pkg.new {
     name = "erg-language-server",
     desc = [[ELS is a language server for the Erg programing language.]],
-    homepage = "https://github.com/erg-lang/erg",
+    homepage = "https://github.com/erg-lang/erg-language-server",
     languages = { Pkg.Lang.Erg },
     categories = { Pkg.Cat.LSP },
     ---@async
     ---@param ctx InstallContext
     install = function(ctx)
-        local repo = "erg-lang/erg"
+        local repo = "erg-lang/erg-language-server"
         local asset_file = _.coalesce(
-            _.when(platform.is.mac_x64, "erg-x86_64-apple-darwin.tar.gz"),
-            _.when(platform.is.mac_arm64, "erg-aarch64-apple-darwin.tar.gz"),
-            _.when(platform.is.linux_x64_gnu, "erg-x86_64-unknown-linux-gnu.tar.gz"),
-            _.when(platform.is.win_x64, "erg-x86_64-pc-windows-msvc.zip")
+            _.when(platform.is.mac_x64, "els-x86_64-apple-darwin.tar.gz"),
+            _.when(platform.is.mac_arm64, "els-aarch64-apple-darwin.tar.gz"),
+            _.when(platform.is.linux_arm64_gnu, "els-aarch64-unknown-linux-gnu.tar.gz"),
+            _.when(platform.is.linux_x64_gnu, "els-x86_64-unknown-linux-gnu.tar.gz"),
+            _.when(platform.is.linux_x64, "els-x86_64-unknown-linux-musl.tar.gz"),
+            _.when(platform.is.win_x64, "els-x86_64-pc-windows-msvc.zip")
         )
         platform.when {
             unix = function()
@@ -27,7 +29,7 @@ return Pkg.new {
                         asset_file = asset_file,
                     })
                     .with_receipt()
-                ctx:link_bin("erg", "erg")
+                ctx:link_bin("els", "els")
             end,
             win = function()
                 github
@@ -36,7 +38,7 @@ return Pkg.new {
                         asset_file = asset_file,
                     })
                     .with_receipt()
-                ctx:link_bin("erg", "erg.exe")
+                ctx:link_bin("els", "els.exe")
             end,
         }
     end,
diff --git a/lua/mason-registry/erg-language-server/init.lua b/lua/mason-registry/erg/init.lua
similarity index 93%
copy from lua/mason-registry/erg-language-server/init.lua
copy to lua/mason-registry/erg/init.lua
index 92ed0bc..4e54b47 100644
--- a/lua/mason-registry/erg-language-server/init.lua
+++ b/lua/mason-registry/erg/init.lua
@@ -4,8 +4,8 @@ local platform = require "mason-core.platform"
 local _ = require "mason-core.functional"
 
 return Pkg.new {
-    name = "erg-language-server",
-    desc = [[ELS is a language server for the Erg programing language.]],
+    name = "erg",
+    desc = [[A statically typed language that can deeply improve the Python ecosystem.]],
     homepage = "https://github.com/erg-lang/erg",
     languages = { Pkg.Lang.Erg },
     categories = { Pkg.Cat.LSP },
```

**Comment by williamboman** [comment]
@williambotman /fixup

File: lua/mason-registry/erg/init.lua [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+local github = require "mason-core.managers.github"
+local platform = require "mason-core.platform"
+local _ = require "mason-core.functional"
+
+return Pkg.new {
+    name = "erg",
+    desc = [[A statically typed language that can deeply improve the Python ecosystem.]],
+    homepage = "https://github.com/erg-lang/erg",
+    languages = { Pkg.Lang.Erg },
+    categories = { Pkg.Cat.LSP },
```
Comment (williamboman) [inline lua/mason-registry/erg/init.lua:11]:
```suggestion
    categories = { Pkg.Cat.LSP, Pkg.Cat.Compiler },
```

**Comment by williamboman** [comment]
> Generally, Mason doesn't yet support managing runtimes/compilers such as erg, even if they bundle a language server.

I've decided to relax this principle somewhat. I changed this PR to simply add `erg` as a new package, leaving `erg-language-server` as-is. The `erg-language-server` should be removed in the future, but this is currently blocked for [these reasons](https://github.com/williamboman/mason.nvim/pull/777#issuecomment-1361171972).

**Comment by pysan3** [comment]
Hi @williamboman,

I am very sorry for the late response, and thank you so much for working on it.

However, since the [PR to nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/pull/2404) has already been merged upstream, we now have a problem that `els` in mason points to `erg --language-server` in nvim-lspconfig in [mason-lspconfig - mappings - server](https://github.com/williamboman/mason-lspconfig.nvim/blob/faeeb5e02eba0aecdf273a50baa164300af33e77/lua/mason-lspconfig/mappings/server.lua#L43)

The official vscode extension of erg does something like [this](https://github.com/erg-lang/vscode-erg/blob/61a36115b6ff45ef3a54137321d4074ccedfbe15/src/extension.ts#L13) to manage both versions which however requires a `if-else`. Do you know any way to achieve this with nvim-lspconfig or do you have any other solutions?

**Pull request merged**